### DISCUSSION TITLE
Is etcd suitable for frequently changing key values?

### DISCUSSION THREAD

**Discussion by AuroraTwinkle** [etcd-io/etcd#16520] [Q&A] [answered]
I want to know that whether etcd is suitable for frequently changing key values?thank you!

**Comment by dev-vaayen** [comment, marked as answer]
Yes it is - You can easily change values of keys using `etcdctl put <key> <value>`.

Let's say you have created a key using `etcdctl put name XYZ`. When you run `etcdctl get name` for retrieving the value of the name key, you will see that the value is XYZ. But if after this, you think you need to update the value of the name key, you would just have to run `etcdctl put name newXYZ`.

Now when you run `etcdctl get name` again, you will be displayed the most recent value of the key. You can also keep track of the previous values of a key.

**Comment by jmhbnz** [comment]
Hey @AuroraTwinkle - Thanks for your question.  Without more detail I can't get into any specifics, but in general, as mentioned by @TheUncertainOracle above, etcd can certainly be used for frequently changing values.

Depending on how often these values are changing please be aware of etcd periodic maintenance procedures and tuning: https://etcd.io/docs/v3.5/op-guide/maintenance

**Reply by AuroraTwinkle** [reply]
@jmhbnz @TheUncertainOracle Thanks for your reply！In my scene, there are a total of 200 keys, occupying 1.2Mb of space. Each key is modified once a second, and I found that my db size has reached around 3GB after compact every 4 hours. I think this is too incredible, is it because the keys are modified too frequently? Do you have any solutions? Thank you

**Reply by jmhbnz** [reply]
With etcd keep all old key revisions until compaction you would need to tune compaction settings if you want to keep database size under a certain limit. Refer to auto compaction settings here https://etcd.io/docs/v3.5/op-guide/maintenance/#history-compaction-v3-api-key-value-database
